src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp

```diff
@@ -677,6 +677,74 @@ class npc_greengill_slave : public CreatureScript
     };
 };
 
+/*######
+## npc_irespeaker
+######*/
+
+enum Spells
+{
+    SPELL_FEL_FIREBALL = 35913,
+    SPELL_CURSE_OF_WEAKNESS = 18267,
+    SPELL_FEL_CONSUMPTION = 45023,
+};
+
+struct npc_irespeaker : public ScriptedAI
+{
+    npc_irespeaker(Creature* creature) : ScriptedAI(creature), _scheduler() {  }
+
+    void JustEngagedWith(Unit* /*who*/ ) override
+    {
+        ScheduleFelFireball(); // Start with Fel Fireball if initially out of melee range
+        ScheduleCurseOfWeakness(); // Start with Curse of Weakness if in melee range
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        me->CastSpell(me, SPELL_FEL_CONSUMPTION, true);
+        _scheduler.CancelAll(); // Cancel all tasks on death
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _scheduler.Update(diff); // Update the TaskScheduler
+
+        if (!UpdateVictim())
+            return;
+
+        if (me->IsWithinMeleeRange(me->GetVictim()))
+            DoMeleeAttackIfReady();
+        else
+            me->GetMotionMaster()->MoveChase(me->GetVictim(), 25.0f);
+    }
+
+private:
+    TaskScheduler _scheduler;
+
+    void ScheduleCurseOfWeakness()
+    {
+        _scheduler.Schedule(0s, [this](TaskContext context)
+        {
+            if (me->IsInCombat() && me->IsWithinMeleeRange(me->GetVictim()))
```

Comment:
No point checking if the creature is in combat, timers shouldnt update if they are not